pandas/core/arrays/categorical.py

```diff
@@ -481,6 +482,14 @@ def _constructor(self) -> type[Categorical]:
     def _from_sequence(cls, scalars, *, dtype: Dtype | None = None, copy=False):
         return Categorical(scalars, dtype=dtype, copy=copy)
 
+    @overload
+    def astype(self, dtype: NpDtype, copy: bool = True) -> np.ndarray:
```

Comment:
i think with dt64/td64 could return DTA/TDA?

Reply:
Don't see that:
```python
>>> s = pd.Series(pd.to_datetime(["2021-03-15 12:05", "2021-04-05 5:10"]), dtype
="category")
>>> s
0   2021-03-15 12:05:00
1   2021-04-05 05:10:00
dtype: category
Categories (2, datetime64[ns]): [2021-03-15 12:05:00, 2021-04-05 05:10:00]
>>> s.dtype.categories.dtype
dtype('<M8[ns]')
>>> type(s.astype(s.dtype.categories.dtype).values)
<class 'numpy.ndarray'>
```